packages/massa-web3/src/web3/BaseClient.ts

```diff
@@ -192,34 +189,34 @@ export class BaseClient {
     }
 
     try {
-      resp = await axios.post(
-        this.getProviderForRpcMethod(resource).url,
-        body,
-        requestHeaders
-      )
-    } catch (ex) {
+      resp = await fetch(this.getProviderForRpcMethod(resource).url, {
+        headers: fetchRequestHeaders,
+        body: JSON.stringify(body),
+        method: 'POST',
+      })
+
+      const responseData: JsonRpcResponseData<T> = await resp.json()
+
+      if (responseData.error) {
+        return {
+          isError: true,
+          result: null,
+          error: new Error(responseData.error.message),
+        } as JsonRpcResponseData<T>
+      }
+
       return {
-        isError: true,
-        result: null,
-        error: new Error('JSON.parse error: ' + String(ex)),
+        isError: false,
+        result: responseData.result as T,
+        error: null,
       } as JsonRpcResponseData<T>
```

Comment:
You can simplify your code by using easy to read ternary operators:
```suggestion
return {
  isError: !!responseData.error,
  result: responseData.error ? null : responseData.result as T,
  error: responseData.error ? new Error(responseData.error.message) : null,
} as JsonRpcResponseData<T>;
```